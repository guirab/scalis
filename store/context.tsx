"use client";

import React, { ReactNode, createContext, useState } from "react";

export const AccountsContext: React.Context<ContextType> = createContext(
  {} as ContextType
);

export const AccountsProvider = ({ children }: { children: ReactNode }) => {
  const [account, setAccount] = useState<AccountType>({} as AccountType);
  const [accounts, setAccounts] = useState<AccountType[]>([] as AccountType[]);

  return (
    <AccountsContext.Provider
      value={{
        account,
        setAccount,
        accounts,
        setAccounts,
      }}
    >
      {children}
    </AccountsContext.Provider>
  );
};
