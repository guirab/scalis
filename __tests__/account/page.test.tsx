import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { AccountsContext } from "../../store/context";
import UserAccount from "../../app/account/page";
import {
  Mock,
  afterEach,
  beforeAll,
  describe,
  expect,
  test,
  vitest,
} from "vitest";
import { useRouter } from "next/navigation";

describe("UserAccount", () => {
  afterEach(() => {
    cleanup();
  });
  beforeAll(() => {
    vitest.mock("next/navigation", () => ({
      useRouter: () => ({
        back: vitest.fn(),
      }),
    }));
    vitest.mock("react", async () => {
      return {
        ...(await vitest.importActual("react")),
        useContext: () => ({
          account: {
            username: "testuser",
            checking: 1000,
            savings: 2000,
          },
          setAccount: vitest.fn(),
        }),
      };
    });
  });
  test("snapshot", () => {
    const { container } = render(<UserAccount />);
    expect(container).toMatchSnapshot();
  });

  test("renders account information correctly", () => {
    const account: any = {
      username: "testuser",
      checking: 100000,
      savings: 200000,
    };

    render(
      <AccountsContext.Provider value={{ ...account }}>
        <UserAccount />
      </AccountsContext.Provider>
    );

    expect(screen.getByText("testuser's Account")).not.toBeNull();
    expect(screen.getByText("Checking Balance: $1,000.00")).not.toBeNull();
    expect(screen.getByText("Savings Balance: $2,000.00")).not.toBeNull();
  });
});
