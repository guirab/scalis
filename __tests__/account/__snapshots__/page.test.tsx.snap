// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserAccount > snapshot 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center relative p-12 sm:p-24 justify-center sm:justify-start"
  >
    <div
      class="data-[show=false]:hidden flex absolute inset-0 bg-black w-full h-full z-20 justify-center items-start text-6xl p-24"
      data-show="true"
    >
      Loading...
    </div>
    <div
      class="w-full pb-12"
    >
      <span
        class="flex items-center text-2xl cursor-pointer w-fit"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
          />
        </svg>
        Back
      </span>
    </div>
    <h1
      class="text-5xl"
    >
      testuser
      's Account
    </h1>
    <div
      class="w-full flex flex-col mt-4 justify-start gap-y-4"
    >
      <h2
        class="text-2xl bg-blue-800 p-4 rounded-lg"
      >
        Checking Balance: 
        $1,000.00
      </h2>
      <h2
        class="text-2xl bg-blue-800 p-4 rounded-lg"
      >
        Savings Balance: 
        $2,000.00
      </h2>
    </div>
    <h2
      class="text-2xl mt-20 mb-12"
    >
      What would you like to do?
    </h2>
    <div
      class="w-full justify-between flex flex-col sm:flex-row gap-y-4"
    >
      <div
        class="flex flex-col bg-gray-500 p-4 rounded-md h-fit"
      >
        <h2
          class="text-2xl cursor-pointer w-fit"
        >
          Deposit
        </h2>
      </div>
      <div
        class="flex flex-col bg-gray-500 p-4 rounded-md h-fit"
      >
        <h2
          class="text-2xl cursor-pointer w-fit"
        >
          Withdraw
        </h2>
      </div>
      <div
        class="flex flex-col bg-gray-500 p-4 rounded-md h-fit"
      >
        <h2
          class="text-2xl cursor-pointer w-fit"
        >
          Transfer
        </h2>
      </div>
    </div>
  </main>
</div>
`;
